iac-extensions/terraform/src/main/java/org/sonar/iac/terraform/checks/azure/AnonymousAccessToResourceCheck.java

```diff
@@ -0,0 +1,162 @@
+    register(AnonymousAccessToResourceCheck::checkDataFactorLinkSerciceOdata, "azurerm_data_factory_linked_service_odata");
+    register(AnonymousAccessToResourceCheck::checkDataFactorLinkSerciceWebAndSftp, "azurerm_data_factory_linked_service_sftp",
+      "azurerm_data_factory_linked_service_web");
+    register(AnonymousAccessToResourceCheck::checkRedisCache, "azurerm_redis_cache");
+    register(AnonymousAccessToResourceCheck::checkStorageAccount, "azurerm_storage_account");
+    register(AnonymousAccessToResourceCheck::checkStorageContainer, "azurerm_storage_container");
+  }
+
+  /**
+   * Loop through all possible 'auth_settings' block in the resource.
+   * Report if there is not such a block in the resource.
+   */
+  private static void checkResourceAuthSettings(CheckContext ctx, BlockTree resource) {
+    List<BlockTree> authSettings = PropertyUtils.getAll(resource, "auth_settings", BlockTree.class);
+    if (authSettings.isEmpty()) {
+      reportResource(ctx, resource, APP_AUTH_MISSING_MESSAGE);
+    } else {
+      authSettings.forEach(settings -> checkAuthSettings(ctx, settings));
+    }
+  }
+
+  /**
+   * Report if 'enabled' is set to false.
+   * Report if 'enabled' is set to true AND ('unauthenticated_client_action' is missing OR set to "AllowAnonymous")
+   */
+  private static void checkAuthSettings(CheckContext ctx, BlockTree authSettings) {
+    PropertyUtils.get(authSettings, "enabled", AttributeTree.class)
+      .filter(enabled -> TextUtils.isValueFalse(enabled.value()))
+      // report if 'enabled' is set to false
+      .ifPresentOrElse(disabled -> ctx.reportIssue(disabled, DISABLED_AUTH_MESSAGE),
+        // check 'unauthenticated_client_action'
+        () -> PropertyUtils.get(authSettings, "unauthenticated_client_action", AttributeTree.class)
+          .ifPresentOrElse(action -> reportSensitiveValue(ctx, action, "AllowAnonymous", AUTHORIZING_ANONYMOUS_MESSAGE),
+            () -> ctx.reportIssue(authSettings, AUTHORIZING_ANONYMOUS_MESSAGE)));
+  }
+
+  private static void checkApiManagementApi(CheckContext ctx, BlockTree resource) {
+    if (PropertyUtils.isMissing(resource, "openid_authentication")) {
+      reportResource(ctx, resource, API_MANAGEMENT_API_MESSAGE);
+    }
+  }
+
+  /**
+   * Report if 'sign_in' block is missing or 'sign_in->enabled' is set to false
+   */
+  private static void checkApiManagement(CheckContext ctx, BlockTree resource) {
+    PropertyUtils.get(resource, "sign_in", BlockTree.class)
+      .ifPresentOrElse(signIn -> PropertyUtils.get(signIn, "enabled", AttributeTree.class)
+        .ifPresent(enabled -> reportOnFalse(ctx, enabled, API_MANAGEMENT_DISABLED_MESSAGE)),
+        () -> reportResource(ctx, resource, API_MANAGEMENT_MISSING_MESSAGE));
+  }
+
+  private static void checkDataFactorLinkSerciceOdata(CheckContext ctx, BlockTree resource) {
+    if (PropertyUtils.isMissing(resource, "basic_auth")) {
```

Comment:
Should this be "basic_auth" or "basic_authentication"? The Terraform docs mention "basic_authentication": https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_odata